llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -4356,11 +4356,13 @@ SDValue DAGCombiner::visitMULFIX(SDNode *N) {
   return SDValue();
 }
 
-SDValue DAGCombiner::visitMUL(SDNode *N) {
+template <class MatchContextClass> SDValue DAGCombiner::visitMUL(SDNode *N) {
   SDValue N0 = N->getOperand(0);
   SDValue N1 = N->getOperand(1);
   EVT VT = N0.getValueType();
   SDLoc DL(N);
+  bool IsVP = N->isVPOpcode();
```

Comment:
Didn't we previously use `bool UseVP = std::is_same_v<MatchContextClass, VPMatchContext>;` ?

Reply:
Changed to this form.